From ed6d8e6d0aee5c6750e93b54c80f318999fe6a7c Mon Sep 17 00:00:00 2001
From: Christopher Degawa <christopher.degawa@intel.com>
Date: Fri, 21 Jan 2022 11:23:49 -0600
Subject: [PATCH] cmake: install .dll to bin dir

Mainly affects Windows where .dll files are considered runtime files
rather than .so files which are just library files

Signed-off-by: Christopher Degawa <christopher.degawa@intel.com>
---
 Source/Lib/Decoder/CMakeLists.txt | 5 ++++-
 Source/Lib/Encoder/CMakeLists.txt | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/Source/Lib/Decoder/CMakeLists.txt b/Source/Lib/Decoder/CMakeLists.txt
index 1c94eb3e..ca7e51f2 100644
--- a/Source/Lib/Decoder/CMakeLists.txt
+++ b/Source/Lib/Decoder/CMakeLists.txt
@@ -111,7 +111,10 @@ if (BUILD_SHARED_LIBS)
         PRIVATE EB_BUILDING_SHARED_LIBS)
     set(DEC_PKG_CONFIG_EXTRA_CFLAGS " -DEB_DLL")
 endif()
-install(TARGETS SvtAv1Dec DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS SvtAv1Dec
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 
 configure_file(pkg-config.pc.in ${CMAKE_BINARY_DIR}/SvtAv1Dec.pc @ONLY)
 install(FILES ${CMAKE_BINARY_DIR}/SvtAv1Dec.pc DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig")
diff --git a/Source/Lib/Encoder/CMakeLists.txt b/Source/Lib/Encoder/CMakeLists.txt
index e3c059e5..9e618858 100644
--- a/Source/Lib/Encoder/CMakeLists.txt
+++ b/Source/Lib/Encoder/CMakeLists.txt
@@ -142,7 +142,10 @@ if (BUILD_SHARED_LIBS)
         PRIVATE EB_BUILDING_SHARED_LIBS)
     set(ENC_PKG_CONFIG_EXTRA_CFLAGS " -DEB_DLL")
 endif()
-install(TARGETS SvtAv1Enc DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS SvtAv1Enc
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 
 configure_file(pkg-config.pc.in ${CMAKE_BINARY_DIR}/SvtAv1Enc.pc @ONLY)
 install(FILES ${CMAKE_BINARY_DIR}/SvtAv1Enc.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-- 
GitLab

